contracts/ControlledGateway.sol

```diff
@@ -0,0 +1,34 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Pausable} from "@openzeppelin/contracts/security/Pausable.sol";
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+
+contract ControlledGateway is AccessControl, Pausable {
+    bytes32 public constant GOVERNANCE_CONTROLLER =
+        keccak256("GOVERNANCE_CONTROLLER");
+
+    address public token;
+
+    constructor(address _token) {
+        _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
+        _setRoleAdmin(GOVERNANCE_CONTROLLER, DEFAULT_ADMIN_ROLE);
+
+        token = _token;
+    }
+
+    function addGovernanceController(address _newController)
```

Comment:
Can we omit this function since we can just use `grantRole()`?

Reply:
This function has `whenNotPaused` modifier which prevents adding a governor during paused state. 
This makes me think that it might be a better idea to actually override `grantRole` and `revokeRole` functions to use `whenNotPaused` modifier. WDYT?

Reply:
just realized that both `_grantRole` and `_revokeRole ` are private functions in this version of OZ-contracts 🤦🏻‍♂️. 
What do you suggest doing here? one way could be to extract the function body and then paste in the overridden functions.

Reply:
> just realized that both _grantRole and _revokeRole are private functions in this version of OZ-contracts

What version are you referring to? IIUC we're on the latest 4.3.2 release and it looks like [grantRole is public](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/0c4de6721d9668d7b5b2c5a9400fd0b2a5e8de90/contracts/access/AccessControl.sol#L129) in that release.

Reply:
> This makes me think that it might be a better idea to actually override grantRole and revokeRole functions to use whenNotPaused modifier.

I think it is fine to allow ACL updates when the contract is not paused especially since the entity that has the ability to modify the ACL is currently also the entity that can pause/unpause. We would just want to make sure to apply the whenNotPaused modifier to any functions in the gateway that should not be usable if the gateway is paused.